api/provider/aws/load_balancer_delete.go

```diff
@@ -28,23 +28,24 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 	}
 
 	// Check for eventually consistency
-	waitUntilLBDeletedFN := func() (shouldRetry bool, err error) {
+	var err error
+	waitUntilLBDeletedFN := func() (shouldRetry bool) {
```

Comment:
this logic should be its own helper function; it's making logic in this master function difficult to follow by putting it here.